### PR TITLE
ci: Check Pull Request Title

### PR DESCRIPTION
**Pull request by wakamsha**
## Summary

Introduce a GitHub Action called [action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request) to check that the Pull Request title is in the appropriate format.

## References

- [frontend-forward/docs/ADRs/2024-02-05-force-pull-request-title-format-for-oss.md at main · moneyforward/frontend-forward](https://github.com/moneyforward/frontend-forward/blob/main/docs/ADRs/2024-02-05-force-pull-request-title-format-for-oss.md)
- https://github.com/moneyforward/frontend-forward/issues/72

### CODE REVIEW

File: .github/workflows/check_pull_request_title.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Check Pull Request Title
+
+on:
+  pull_request_target:
```
Comment (thinceller) [inline .github/workflows/check_pull_request_title.yml:4]: [memo]
https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#event-triggers によるとOSSのようなforkをベースとしたPRを受け付けるような場合は`pull_request_target`を使うのが良い 👍

File: .github/workflows/check_pull_request_title.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          # Based on the Angular convention
+          # cf. https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines
+          types: |
```
[the remark this answers is not in the document]
Comment (wakamsha) [reply .github/workflows/check_pull_request_title.yml:24]: @thinceller 
In that case, an error will occur and you'll need to specify the default value `feat:` or `fix:`.

[the remark this answers is not in the document]
Comment (thinceller) [reply .github/workflows/check_pull_request_title.yml:24]: Scopes in Conventional Commits Spec are different from types like `feat` or `fix`, they provide additional contextual information such as `feat(ui):` or `fix(api)`.
cf. https://github.com/amannn/action-semantic-pull-request?tab=readme-ov-file#configuration

I mean this file doesn't provide some configuration about scopes, so I guess how this action works for scopes

[the remark this answers is not in the document]
Comment (wakamsha) [reply .github/workflows/check_pull_request_title.yml:24]: @thinceller 
Got it. In this configuration, any word can be specified as `scope`, and it'll work even if it's not specified. Currently I don't think the need to define scope. If you think it's necessary, please raise an issue.

[the remark this answers is not in the document]
Comment (thinceller) [reply .github/workflows/check_pull_request_title.yml:24]: > In this configuration, any word can be specified as `scope`, and it'll work even if it's not specified. Currently I don't think the need to define scope.

That sounds good 👍

**Review by thinceller — APPROVED** [review]

**Pull request merged**